security/patch-basic-auth.yml

```diff
@@ -0,0 +1,21 @@
+#-----------------------------------------------------------------------------------------------------------------
+# Example configuration for securing spinnaker with basic authentication. Not recommended for Production use.
+#
+#-----------------------------------------------------------------------------------------------------------------
+apiVersion: spinnaker.armory.io/v1alpha2
+kind: SpinnakerService
+metadata:
+  name: spinnaker
+spec:
+  spinnakerConfig:
+    profiles:
+      deck:
+        settings-local.js: |
+          window.spinnakerSettings.authEnabled = true;
+      gate:
+        security:
+          basicform:
+            enabled: true
+          user:
+            name: username2replace                        # The basic auth username you want to use.
+            password: password2replace                    # The basic auth password you want to use.
```

Comment:
This field can also be set to something like `encrypted:k8s!n:spin-secrets!k:spin-basic-pass` to be consistent on not using literal secrets in any part of the config, but use encrypted references instead.
Then the `spin-basic-pass` entry can be added to https://github.com/armory/spinnaker-kustomize-patches/blob/master/secrets/secrets-example.env to have an out of the box working config with all secrets isolated.

Reply:
Added as suggested. I did not add this to secrets-example.env as it isn't very often that someone uses basic auth for anything other than a longer-living sandbox. Almost always we see them use SAML or OAUTH.